### PR TITLE
Change to a more relaxed return type phpdoc

### PR DESCRIPTION
**Pull request by nevvermind**
See https://github.com/magento/magento2/issues/1578

### CODE REVIEW

File: app/code/Magento/Sales/Api/Data/OrderItemInterface.php [hunk 1]
```diff
@@ -856,7 +856,7 @@ public function getProductId();
     /**
      * Gets the product options for the order item.
      *
-     * @return string[] Array of product options.
+     * @return array Array of product options.
```
Comment (orlangur) [inline app/code/Magento/Sales/Api/Data/OrderItemInterface.php]: Won't `array[]|string[]` do the trick?

Comment (nevvermind) [inline app/code/Magento/Sales/Api/Data/OrderItemInterface.php]: Not sure. I guess. 
I saw [this](https://github.com/magento/magento2/blob/077584c99ebb8007cad176c3b9a0144a05c259cd/lib/internal/Magento/Framework/Reflection/DataObjectProcessor.php#L110) and automatically assumed that M2 takes out the last 2 chars from `type[]`, leaving just `type`; haven't thought of `@return type1|type2`. I'll try it.

**Comment by magento-cicd** [comment]
We have automated a Magento Contributor License Agreement verifier for contributions sent to our GitHub projects.
Please see the CLA agreement in the Pull Request comments below.

**Comment by davidalger** [comment]
The method this PR changes the PHPDoc for no longer exists in 2.0 RC. Closing

**Pull request closed without merging**